src/licensedcode/models.py

```diff
@@ -1428,13 +1424,35 @@ class BasicRule:
             'Mutually exclusive from any is_license_* flag')
     )
 
+    is_required_phrase = attr.ib(
+        default=False,
+        repr=False,
+        metadata=dict(
+            help='True if this is rule text is a required phrase. '
```

Comment:
I am still unsure about this field... is this when the whole rule text is a required phrase? Because looking at this again now, I wonder if this would not be simpler to generate the flag when we index a rule based on this fact.

Reply:
There are a couple reasons why we are doing it this way, explicitly with a flag, instead of computing these on the fly:

1. We are not generating these required phrase rules while loading the index from the rules, this is instead an one time operation and a thing that only runs on tests or new rule additions, where we go thorough the rules collect required phrases and mark them in other rules (while also saving them, by creating a new file, or already marking an existing rule with the flag). If we start generating this on the fly while creating the index we are making the index creation step more expensive (we now only collect required phrases from a single rule, but if we do this we have to also add them as rules to the main index) and it will take even more time.
2. Today the process to create required phrase rules is not fully automated at first, we need to review all the newly created required phrase rules and then mark some origin rules with `skip_collecting_required_phrases: yes` so that some rule-specific required phrases are not created as required phrases. This process will become much more abstract and non-intuitive, and would be much more difficult to debug.
3. The license detection process update will also use these new required phrase rules. In cases where we don't have good matches, we will try to do a license detection just filtering required phrase rules from all the matches, and this would involve looking at all matches which has `is_required_phrase` as True, therefore making this flag necessary.

Reply:
OK.Thanks!